src/WebJobs.Extensions.DurableTask/EntityScheduler/Entity.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Statically accessible context for entity operations.
+    /// </summary>
+    public static class Entity
+    {
+        private static readonly AsyncLocal<IDurableEntityContext> EntityContext
+            = new AsyncLocal<IDurableEntityContext>();
```

Comment:
I appreciate that `AsyncLocal<T>` enables access to the current entity context from within the typed actor implementation. However, like with all static variables, what is the impact to unit testing? Will users be able to mock this?

Reply:
I am not really familiar with the problem or how one would solve that in general. 

I suppose we could use dependency injection as an alternative mechanism for people who want to unit test their code. This is what Orleans does. Not sure how we pull that off though. As far as I know support for dependency injection in AF was just announced, but it may take some work to figure out how to hook it up with the DF piece.

Reply:
One simple option would be to make `Entity.Current` settable (that or `Entity.SetContext(...))`. Someone writing a unit test could then set the static context themselves to avoid null-references. But of course this has the negative side effect of encouraging developers to set this property in production code as well - so we would have to discourage this in documentation.

Another option would be to allow the entity constructor to take an `IDurableEntityContext` parameter. If present, we could call that constructor and the user could save a reference to it for use in their operation handling logic. That solves the problem as well, though requires the user to write more boilerplate code if they want to enable unit testing.

Reply:
I suppose I like the first suggestion better. The second one has some pitfalls (e.g. users store IDurableEntityContext in a field then get SerializationException because they did not mark it as `[JsonIgnore]`) and is a bit counterintuitive since the context is not really part of the actor state.

Perhaps provide `Entity.SetMockContext(…)` to indicate that it is for testing purposes? We could even throw an exception if users use it in a real situation, although I am not sure we need to worry too much about it.

Reply:
I've used this approach in the past, and I like your implementation of it since it's able to distinguish between a real context object and a mock one.